packages/workflow/src/RoutingNode.ts

```diff
@@ -209,17 +211,29 @@ export class RoutingNode {
 
 				returnData.push(...responseData);
 			} catch (error) {
-				if (get(this.node, 'continueOnFail', false)) {
-					returnData.push({ json: {}, error: error.message });
+				interface AxiosError extends NodeError {
+					isAxiosError: boolean;
+					description: string | undefined;
+					response?: { status: number };
+				}
+
+				const routingError = error as AxiosError;
+
+				if (this.node.continueOnFail) {
+					returnData.push({ json: {}, error: routingError });
```

Comment:
this changes the `error` in `returnData` from `string` to an `object`. Is there a way to check if this won't break anything?

Reply:
I noticed this - `INodeExecutionData["error"]` is `NodeApiError | NodeOperationError`, not `string`, so in theory `error.message` has been wrong all along.

Afaik `RoutingNode` is used by declarative nodes, e.g. the n8n node, which seems to continue to [process errors correctly](https://share.cleanshot.com/9ds3Xvz8).

Reply:
Two things:
1. Fixed a misreference with `error.cause`, else `isAxiosError` and `response` were `undefined`.
2. See [before](https://share.cleanshot.com/kY1b9yw7) and [after](https://share.cleanshot.com/qv8MdR14) on continue on fail for declarative nodes - this PR fixes the issue.